clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/AbstractEcsCachingAgent.java

```diff
@@ -82,7 +82,10 @@ public String getProviderName() {
   public CacheResult loadData(ProviderCache providerCache) {
     String authoritativeKeyName = getAuthoritativeKeyName();
 
-    AmazonECS ecs = amazonClientProvider.getAmazonEcs(accountName, awsCredentialsProvider, region);
+    AmazonECS ecs = null;
+    if(region!=null && region.length()!=0){
+      ecs = amazonClientProvider.getAmazonEcs(accountName, awsCredentialsProvider, region);
+    }
     List<T> items = getItems(ecs, providerCache);
```

Comment:
What if region == null or region.length = 0?  We'd get an NPE in `getItems(ecs, providerCache)`

Reply:
Only CachingAgent that doesn't have a region is `IamRoleCachingAgent` every other agents takes in a region in their constructor - I can make a check in their constructors.